catalog/services_state_test.go

```diff
@@ -205,30 +205,31 @@ func Test_ServicesStateWithData(t *testing.T) {
 			})
 
 			Convey("Doesn't retransmit when the state is the same", func() {
-				state.Broadcasts = make(chan [][]byte, 1)
+				state.Broadcasts = make(chan [][]byte, 2)
 				state.AddServiceEntry(svc)
+
+				<-state.Broadcasts
+
 				svc.Updated = svc.Updated.Add(1 * time.Second)
 				state.AddServiceEntry(svc)
 
-				pendingBroadcast := false
-				select {
-				case <-state.Broadcasts:
-					pendingBroadcast = true
-				default:
-				}
-				So(pendingBroadcast, ShouldBeFalse)
+				time.Sleep(5 * time.Millisecond)
+
+				So(len(state.Broadcasts), ShouldEqual, 0)
```

Comment:
@relistan I gave this test some more thought and I think the underlying mechanism might not work as intended, or this test doesn't make sense any more. If you try to run it now with my changes, it will fail saying that `state.Broadcasts` has one message in it instead of zero.

It looks like [`Service.Invalidates`](https://github.com/Nitro/sidecar/blob/master/service/service.go#L58) returns `true` if `svc.Updated` is more recent (makes sense), so [this code path](https://github.com/Nitro/sidecar/blob/master/catalog/services_state.go#L280) will get followed.

So, given that we have different host names ([`os.Hostname()`](https://github.com/Nitro/sidecar/blob/master/catalog/services_state.go#L89) versus [`anotherHostname`](https://github.com/Nitro/sidecar/blob/master/catalog/services_state_test.go#L90)), `ServicesState.retransmit` will send a second broadcast.

Was there supposed to be some sort of check in the code to avoid the second broadcast? The test description seems to imply that.